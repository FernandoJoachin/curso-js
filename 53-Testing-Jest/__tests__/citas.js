import Citas from "../js/classes/Citas";

describe('Testing de la clase Citas', () => {
    const citas = new Citas();
    const id = Date.now();

    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hunter',
            propietario: 'Fernando',
            telefono: '1234567890',
            fecha: '22-03-2024',
            hora:'2:57',
            sintomas: 'Fiebre'
        }
        citas.agregarCita(citaObj);
        expect(citas).toMatchSnapshot();
    });

    test('Actualizar una cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Eli',
            propietario: 'Fernando',
            telefono: '1234567890',
            fecha: '22-03-2024',
            hora:'2:57',
            sintomas: 'Fiebre'
        }
        citas.editarCita(citaActualizada);
        expect(citas).toMatchSnapshot();
    });

    test('Eliminar una cita', () => {
        citas.eliminarCita(id);
        expect(citas).toMatchSnapshot();
    });

   
})