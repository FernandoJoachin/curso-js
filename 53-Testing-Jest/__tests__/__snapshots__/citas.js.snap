// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing de la clase Citas Actualizar una cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "22-03-2024",
      "hora": "2:57",
      "id": 1711098422287,
      "mascota": "Eli",
      "propietario": "Fernando",
      "sintomas": "Fiebre",
      "telefono": "1234567890",
    },
  ],
}
`;

exports[`Testing de la clase Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "22-03-2024",
      "hora": "2:57",
      "id": 1711098422287,
      "mascota": "Hunter",
      "propietario": "Fernando",
      "sintomas": "Fiebre",
      "telefono": "1234567890",
    },
  ],
}
`;

exports[`Testing de la clase Citas Eliminar una cita 1`] = `
Citas {
  "citas": [],
}
`;
